README.md

```diff
@@ -1,73 +1,203 @@
+
+# Usage
+
+## Installing the Package
 
-## Installation
 ```bash
-pip3 install realtime==1.0.2
+pip3 install realtime==2.0.0
 ```
 
-## Installation from source 
-```bash
-pip3 install -r requirements.txt
-python3 usage.py 
+## Creating a Channel
 
+```python
+import asyncio
+from typing import Optional
+from realtime.client import RealtimeClient
+from realtime.channel import RealtimeSubscribeStates
+
+client = RealtimeClient(REALTIME_URL, API_KEY)
+channel = client.channel('test-channel')
+
+def _on_subscribe(status: RealtimeSubscribeStates, err: Optional[Exception]):
+    if status == RealtimeSubscribeStates.SUBSCRIBED:
+        print('Connected!')
+    elif status == RealtimeSubscribeStates.CHANNEL_ERROR:
+        print(f'There was an error subscribing to channel: {err.message}')
+    elif status == RealtimeSubscribeStates.TIMED_OUT:
+        print('Realtime server did not respond in time.')
+    elif status == RealtimeSubscribeStates.CLOSED:
+        print('Realtime channel was unexpectedly closed.')
+
+await channel.subscribe(_on_subscribe)
+
+# Listen for all incoming events, often the last thing you want to do.
+await client.listen()
 ```
 
-## Quick Start 
+### Notes:
+
+- `REALTIME_URL` is `'ws://localhost:4000/socket'` when developing locally and `'wss://<project_ref>.supabase.co/realtime/v1'` when connecting to your Supabase project.
+- `API_KEY` is a JWT whose claims must contain `exp` and `role` (existing database role).
+- Channel name can be any `string`.
+
+## Broadcast
+
+Your client can send and receive messages based on the `event`.
+
 ```python
-from realtime.connection import Socket
+# Setup...
+
+channel = client.channel(
+    "broadcast-test", {"config": {"broadcast": {"ack": False, "self": False}}}
+)
+
+await channel.on_broadcast("some-event", lambda payload: print(payload)).subscribe()
+await channel.send_broadcast("some-event", {"hello": "world"})
+```
+
+### Notes:
 
-def callback1(payload):
-    print("Callback 1: ", payload)
+- Setting `ack` to `true` means that the `channel.send` promise will resolve once server replies with acknowledgement that it received the broadcast message request.
+- Setting `self` to `true` means that the client will receive the broadcast message it sent out.
+- Setting `private` to `true` means that the client will use RLS to determine if the user can connect or not to a given channel.
 
-def callback2(payload):
-    print("Callback 2: ", payload)
+## Presence
 
-if __name__ == "__main__":
-    URL = "ws://localhost:4000/socket/websocket"
-    s = Socket(URL)
-    s.connect()
+Your client can track and sync state that's stored in the channel.
+
+```python
+# Setup...
+
+channel = client.channel(
+    "presence-test",
+    {
+        "config": {
+            "presence": {
+                "key": ""
+            }
+        }
+    }
+)
+
+channel.on_presence_sync(lambda: print("Online users: ", channel.presenceState()))
```

Comment:
Can't seem to find `presenceState`. Have we implemented it yet?

```suggestion
channel.on_presence_sync(lambda: print("Online users: ", channel.presenceState()))
```

Reply:
nice catch, thanks